.github/workflows/applications.yml

```diff
@@ -0,0 +1,282 @@
+      with:
+        version: "4.12.1"
+        architecture: "x64"
+        distribution: "osx-10.16"
+    - name: Print Z3 version
+      shell: bash
+      run: z3 -version
+   
+    # Haskell Setup
+    - name: Set permissions for .ghcup (ubuntu)
+      if: startsWith(matrix.os, 'ubuntu-')
+      run: sudo chown -R $USER /usr/local/.ghcup
+    - name: Install GHC and Cabal
+      uses: haskell-actions/setup@v2
+      with:
+         ghc-version: ${{ matrix.ghc }}
+         cabal-version: ${{ matrix.cabal }}
+    - name: Confirm GHC and Cabal installation
+      run: |
+        ghc --version
+        cabal --version
+
+    # Project Setup
+    - name: Create cabal.project.local
+      shell: bash
+      run: |
+        cat > cabal.project.local <<EOF
+        documentation: False
+        package pact
+          tests: True
+          benchmarks: True
+          documentation: False
+          optimization: 1
+          flags: ${{ matrix.flags }} +cryptonite-ed25519
+        extra-include-dirs:
+          /opt/local/include
+          /usr/local/opt/openssl/include
+        extra-lib-dirs:
+          /opt/local/lib
+          /usr/local/opt/openssl/lib/
+        EOF
+    - name: Extend cabal.project.local for GHC-9.0.2
+      if: "startsWith(matrix.ghc, '9')"
+      shell: bash
+      run: |
+        cat >> cabal.project.local <<EOF
+        package pact
+          ghc-options: -Wwarn -Wunused-packages
+        EOF
+    - name: Add check for unused packages
+      shell: bash
+      run: |
+        cat >> cabal.project.local <<EOF
+        package pact
+          ghc-options: -Wunused-packages
+        EOF
+    - name: Print cabal.project.local
+      shell: bash
+      run: cat cabal.project.local
+    - uses: actions/cache@v3
+      name: Cache dist-newstyle
+      with:
+        path: |
+          ~/.cabal/packages
+          ~/.cabal/store
+          dist-newstyle
+        key: ${{ matrix.os }}-${{ matrix.ghc }}-4-cabal
+
+    # Build
+    - name: Update package database
+      shell: bash
+      run: cabal update
+    - name: Display outdated packages
+      run: cabal outdated
+    - name: Configure build
+      run: |
+        cabal build --dry-run
+        cabal freeze
+    - name: Sync from cabal cache
+      if: matrix.cabalcache == 'true'
+      uses: larskuhtz/cabal-cache-action@4b537195b33898fcd9adc62cee2a44986fd7b1b6
+      with:
+        bucket: "kadena-cabal-cache"
+        region: "us-east-1"
+        folder: "packages/${{ matrix.os }}"
+        aws_access_key_id: "${{ secrets.kadena_cabal_cache_aws_access_key_id }}"
+        aws_secret_access_key: "${{ secrets.kadena_cabal_cache_aws_secret_access_key }}"
+    - name: Build dependencies
+      shell: bash
+      run: cabal build --only-dependencies
+    - name: Build
+      shell: bash
+      run: cabal build
+    - name: Test - non-ARM
+      if: "!startsWith(matrix.os, 'macos-')"
+      shell: bash
+      run: cabal run tests
+    # - name: Test - ARM
```

Comment:
Is this still needed?

Reply:
right, that's outdated, removing

Reply:
for reference the old pact repo had arm special builds because of a weird test that only worked on amd64 and pact-5's cabal run tests runs a different suite that works universally